ai-assistant/src/modal.tsx

```diff
@@ -353,6 +354,66 @@ export default function AIPrompt(props: {
     [aiManager, updateHistory]
   );
 
+  const handleOperationFailure = React.useCallback(
+    (error: any, operationType: string, resourceInfo?: any) => {
+      // Determine the operation type from the error or method
+      let operation = 'operation';
+      if (operationType) {
+        switch (operationType.toLowerCase()) {
+          case 'post':
+            operation = 'creation';
+            break;
+          case 'put':
+          case 'patch':
+            operation = 'update';
+            break;
+          case 'delete':
+            operation = 'deletion';
+            break;
+          default:
+            operation = 'operation';
+        }
+      }
+
+      // Extract error details
+      const errorMessage = error?.message || error?.error || 'Unknown error occurred';
+      const statusCode = error?.status || error?.statusCode;
+
+      // Build error content
+      let errorContent = `Resource ${operation} failed: ${errorMessage}`;
```

Comment:
This will be complicated to handle when we introduce translations for plugins.

Reply:
updated